smarts/sstudio/generators.py

```diff
@@ -368,3 +371,36 @@ def _resolve_log_dir(self, log_dir):
             log_dir = make_dir_in_smarts_log_dir("_duarouter_routing")
 
         return os.path.abspath(log_dir)
+
+
+def tripwritexml(resolve_route, traffic, doc):
```

Comment:
A couple minor things here:
- The name should probably match the style convention -- maybe `write_trip_xml`?
- I don't think this should be a top-level function. It should either be encapsulated in the `TrafficGenerator` class or the code should be inlined where it's called (I don't have a strong opinion either way).

Reply:
I will change the name into write_trip_xml and encapsulated it into a class method using self.resolve_route instead of passing resolve_route as an argument.